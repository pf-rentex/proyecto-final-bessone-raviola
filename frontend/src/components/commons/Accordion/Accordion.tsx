import React, { useRef, useState } from "react";
import { FaChevronCircleDown } from "react-icons/all";

interface AccordionProps {
  title: string;
  children: React.ReactNode;
}

export const Accordion: React.FC<AccordionProps> = ({
  title,
  children = <></>,
}) => {
  const [active, setActive] = useState(false);
  const [height, setHeight] = useState("0px");
  const [rotate, setRotate] = useState("transform duration-300 ease");

  const contentSpace = useRef(null);

  function toggleAccordion() {
    setActive(!active);
    // @ts-ignore
    setHeight(active ? "0px" : `${contentSpace.current.scrollHeight}px`);
    setRotate(
      active
        ? "transform duration-300 ease"
        : "transform duration-300 ease rotate-180"
    );
  }

  return (
    <div className="flex flex-col">
      <button
        className="box-border appearance-none cursor-pointer focus:outline-none flex items-center justify-between"
        onClick={toggleAccordion}
      >
        <p className="inline-block text-footnote light">{title}</p>
        <FaChevronCircleDown
          className={`${rotate} inline-block text-alt w-7 h-7`}
        />
      </button>
      <div
        ref={contentSpace}
        style={{ maxHeight: `${height}` }}
        className="overflow-hidden transition-max-height duration-300 ease-in-out"
      >
        <div className="pt-5">{children}</div>
      </div>
    </div>
  );
};
