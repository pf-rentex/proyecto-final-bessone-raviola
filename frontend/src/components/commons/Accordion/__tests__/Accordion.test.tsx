import React from "react";
import {render} from "@testing-library/react";
import {Accordion} from "../Accordion";

describe("Accordion tests", () => {
  const content = (
      <ul className="font-thin space-y-5">
        <li className="flex justify-between">Opci贸n 1</li>
        <li className="flex justify-between">Opci贸n 2</li>
      </ul>
  );
  const getWrapper = () =>
      render(<Accordion title="Mis inmuebles">{content}</Accordion>);

  test("renders component", () => {
    const wrapper = getWrapper();

    expect(wrapper.container).toBeInTheDocument();
    expect(wrapper.container).toMatchSnapshot();
  });

  test("Contains title element", () => {
    const {getByText} = getWrapper();

    getByText("Mis inmuebles");
  });

  test("contains children elements", () => {
    const wrapper = getWrapper();

    wrapper.getByTestId("button");
    wrapper.getByText("Opci贸n 1");
    wrapper.getByText("Opci贸n 2");
  });
});
