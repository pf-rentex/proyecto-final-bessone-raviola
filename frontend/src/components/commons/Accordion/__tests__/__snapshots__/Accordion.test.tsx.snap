// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion tests renders component 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <button
      class="box-border appearance-none cursor-pointer focus:outline-none flex items-center justify-between"
      data-testid="button"
    >
      <p
        class="inline-block text-footnote light"
      >
        Mis inmuebles
      </p>
      <svg
        class="transform duration-300 ease inline-block text-alt w-7 h-7"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zM273 369.9l135.5-135.5c9.4-9.4 9.4-24.6 0-33.9l-17-17c-9.4-9.4-24.6-9.4-33.9 0L256 285.1 154.4 183.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L239 369.9c9.4 9.4 24.6 9.4 34 0z"
        />
      </svg>
    </button>
    <div
      class="overflow-hidden transition-max-height duration-300 ease-in-out"
      style="max-height: 0px;"
    >
      <div
        class="pt-5"
      >
        <ul
          class="font-thin space-y-5"
        >
          <li
            class="flex justify-between"
          >
            Opción 1
          </li>
          <li
            class="flex justify-between"
          >
            Opción 2
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
